src/common/docker-compose/index.ts

```diff
@@ -672,9 +672,14 @@ export class DockerComposeUtils {
           const state = container_state.State.toLowerCase();
           const health = container_state.Health.toLowerCase();
 
-          const bad_state = state !== 'running';
+          const bad_state = state !== 'running' && container_state.ExitCode !== 0;
           const bad_health = health === 'unhealthy';
 
+          // Stop watching when container exited successfully.
+          if (state === 'exited' && container_state.ExitCode === 0) {
+            return true;
```

Comment:
We should return `restarted` here instead of `true` - kind of complicated but the result of this is looked at when `dev` is doing it's cleanup and if a container restarted _while_ cleanup is happening, we need to do an additional step, so if we just return `restarted` here that'll do the right thing in weird edge cases

Reply:
This makes sense. Returning `true` will call the `stop` command again.

Reply:
I'll change it to return `restarted`.